### DISCUSSION TITLE
I'm having trouble enabling pylint in python-lsp.

### DISCUSSION THREAD

**Discussion by Sturm0** [techee/geany-lsp#40] [Q&A] [answered]
It seems that in addition to installing pylint (which I have installed in version 3.0.3) I have to enable an option called pylsp.plugins.pylint.enabled (https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md)

From lsp.conf I see that this section seems to talk about how I could do it:
```
# File containing initialization options of the server. The server is
# automatically restarted when this file is modified from within Geany
initialization_options_file=/home/some_user/init_options.json
```
So I changed the configuration like this for the python section:
```
[Python]
cmd=pylsp
initialization_options_file=/home/my_name/.config/init_options.json
```

Now, regarding the content of init_options.json it seems that it is not something standard that I have to put there but that it depends on the ide: https://github.com/python-lsp/python-lsp-server/discussions/176
I tried to put this in the file /home/my_name/.config/init_options.json:
```
{
  "settings": {
    "pylsp": {
      "configurationSources": [
        "pycodestyle"
      ],
      "plugins": {
        "autopep8": {
          "enabled": false
        },
        "pylint": {
			"enabled" : true
		}
      },
      "rope": {
        "extensionModules": null,
        "ropeFolder": null
      }
    }
  }
}
```
I tried to see if I had any luck, but it didn't work.
I also tried reading https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_configuration
as it is mentioned in:
```
# JSON file containing formatting options defined in
# https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#formattingOptions
# e.g. { "tabSize": 4, "insertSpaces": false }. Supported only by some language
```
But I also didn't end up understanding how I should put together the message

How should I do it?

**Comment by techee** [comment, marked as answer]
If I remember correctly, I also had trouble getting the initialization options sent to the server for pylsp (it worked for other servers though).

What you might try is to drop
```
  "settings": {
    "pylsp": {
    }
  }
```
because these could just be vscode settings specifying which server to configure. So I'd suggest trying
```
{
      "configurationSources": [
        "pycodestyle"
      ],
      "plugins": {
        "autopep8": {
          "enabled": false
        },
        "pylint": {
			"enabled" : true
		}
      },
      "rope": {
        "extensionModules": null,
        "ropeFolder": null
      }
}
```

or possibly keeping `pylsp` and dropping `settings` only.

Also check https://github.com/python-lsp/python-lsp-server/discussions/176#discussioncomment-2395159 - maybe it should be configured in the second format and you use the whole strings from

https://github.com/python-lsp/python-lsp-server/blob/develop/CONFIGURATION.md

as keys.

Let me know if it works.

**Reply by Sturm0** [reply]
I just tried:
```
{
      "configurationSources": [
        "pycodestyle"
      ],
      "plugins": {
        "autopep8": {
          "enabled": false
        },
        "pylint": {
			"enabled" : true
		}
      },
      "rope": {
        "extensionModules": null,
        "ropeFolder": null
      }
}
```
No luck.
With:
```
{
    "pylsp": {
      "configurationSources": [
        "pycodestyle"
      ],
      "plugins": {
        "autopep8": {
          "enabled": false
        },
        "pylint": {
			"enabled" : true
		}
      },
      "rope": {
        "extensionModules": null,
        "ropeFolder": null
      }
    }
}
```
IT WORKS !!!!!! THANKS

**Reply by techee** [reply]
Good to hear! If you run into any other problems, just let me know.
